annotation-utils/src/main/java/edu/isi/vista/annotationutils/ParseEventLogs.kt

```diff
@@ -229,7 +237,9 @@ private fun parseProjectEvents(logEvents: List<JsonNode>, username: String): Pai
     for (event in logEvents) {
         val inceptionEventType = event.get("event").toString().removeSurrounding("\"")
         val documentName = event.get("document_name")?.toString()?.removeSurrounding("\"")
-        val user = event.get("user")?.toString()?.removeSurrounding("\"")
+        val logUser = event.get("user")?.toString()?.removeSurrounding("\"")
+        val mappedUser = usernameMap.get(logUser)?.toString()?.removeSurrounding("\"")
+        val user = mappedUser ?: logUser
```

Comment:
Just curious - how does this `?:` work?

Reply:
It's basically short for "if x is not null then x else y"
https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator

Reply:
So here, if `mappedUser` is not null, `user` will be `mappedUser`, else it will be `logUser`.